src/Npgsql/CrateDbDatabaseInfo.cs

```diff
@@ -0,0 +1,203 @@
+                    mappings.Add(t.PgTypeName, t);
+            }
+        }
+
+        /// <summary>
+        /// Returns a boolean value that signals if transactions are supported by the current database.
+        /// </summary>
+        public override bool SupportsTransactions => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD command is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscard => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD TEMP subcommand is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscardTemp => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD SEQUENCES subcommand is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscardSequences => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the CLOSE ALL command is supported by the current database.
+        /// </summary>
+        public override bool SupportsCloseAll => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the UNLISTEN command is supported by the current database.
+        /// </summary>
+        public override bool SupportsUnlisten => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports range types.
+        /// </summary>
+        public override bool SupportsRangeTypes => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports advisory lock functions.
+        /// </summary>
+        public override bool SupportsAdvisoryLocks => false;
+        /// <summary>
+        /// Reports whether the backend uses the newer integer timestamp representation. 
+        /// CrateDB used floating point timestamps until version &lt;=3.0 and switched to integer datetimes with version &gt;= 3.1.
+        /// </summary>
+        public override bool HasIntegerDateTimes { get; protected set; } = true;
+
+        static IEnumerable<NpgsqlTypeMapping> CrateDbSpecificTypeMappings()
+        {
+            // Map CrateDB varchar type to the Npgsql TextHandler.
+            yield return new NpgsqlTypeMappingBuilder
+            {
+                PgTypeName = "character varying",
+                NpgsqlDbType = NpgsqlDbType.Varchar,
+                DbTypes = new[] { DbType.String, DbType.StringFixedLength, DbType.AnsiString, DbType.AnsiStringFixedLength },
+                ClrTypes = new[] { typeof(string), typeof(char[]), typeof(char) },
+                InferredDbType = DbType.String,
+                TypeHandlerFactory = new TextHandlerFactory()
+            }
+            .Build();
+
+            // Map CrateDB timestampz type to the CrateDbTimestampHandler.
+            yield return new NpgsqlTypeMappingBuilder
```

Comment:
Is this done because you want DateTime to map to `timestamptz` instead of `timestamp`? Can you please provide more context on this, is there no regular `timestamp` on CrateDB?

Reply:
There is a `timestamp` type in CrateDB, hower it is mapped and streamed as `timestamptz` over the Postgres protocol.

pg_catlog.pg_type of CrateDB

```
+------+-------------+----------+---------+-------------+---------+-----------+
|  oid | typbasetype | typdelim | typelem | typname     | typtype | typtypmod |
+------+-------------+----------+---------+-------------+---------+-----------+
| 1043 |           0 | ,        |       0 | varchar     | b       |        -1 |
| 1184 |           0 | ,        |       0 | timestamptz | b       |        -1 |
+------+-------------+----------+---------+-------------+---------+-----------+
```